src/Runner.Worker/ActionManager.cs

```diff
@@ -596,6 +596,12 @@ private async Task BuildActionContainerAsync(IExecutionContext executionContext,
                 }
                 catch (Exception ex)
                 {
+                    // Action download was canceled, due to fail fast strategy. There is no point in retrying anymore as the run is canceled.
+                    if (ex.Message != null && string.Equals(ex.Message, "A task was canceled.", StringComparison.OrdinalIgnoreCase))
```

Comment:
you can do something like 
```
catch (Exception ex) when (!executionContext.CancellationToken.IsCancelled)
```

So you don't have to check the exception message.